pkg/apis/kubermatic/v1/cluster.go

```diff
@@ -1184,9 +1199,18 @@ type BringYourOwnCloudSpec struct{}
 type AWSCloudSpec struct {
 	CredentialsReference *providerconfig.GlobalSecretKeySelector `json:"credentialsReference,omitempty"`
 
-	AccessKeyID          string `json:"accessKeyID,omitempty"`
-	SecretAccessKey      string `json:"secretAccessKey,omitempty"`
-	AssumeRoleARN        string `json:"assumeRoleARN,omitempty"` //nolint:tagliatelle
+	// The Access key ID used to authenticate against AWS.
+	AccessKeyID string `json:"accessKeyID,omitempty"`
+	// The Secret Access Key used to authenticate against AWS.
+	SecretAccessKey string `json:"secretAccessKey,omitempty"`
+	// Defines the ARN for an IAM role that should be assumed when handling resources on AWS. It will be used
+	// to acquire temporary security credentials using an STS AssumeRole API operation whenever creating an AWS session.
+	// required: false
```

Comment:
Is this `required: false` meant to be a comment for the CRD builder? Cause it ain't :grin: The field being `omitempty` already makes it optional, but if you want, you could add `// +optional` to truly signify that.

Reply:
I think I tried to replicate the comment. Yes, `// +optional` would be better fit. I will fix this.  👍